mkdocs/docs/meili/v0.0.3/orchestration_thoughts.md

```diff
@@ -0,0 +1,53 @@
+
+Implementation wise, we need to make sure each run behaves the same and reads one source of truth, every file should be generated into the same `.generated/` file. Then we need to do content hashing and env fingerprints, meaning we read the bytes of important files like the TS models and hash them. This ensures that when we rebuild and the hash is the same, we can quickly keep going and build since we know nothing has changed. Then we formalize each build step as a worker with a clear contract, as in each step should have dependencies that must be completed before we run anything, what counts as inputs, the output it promises to write, and how we run everything. Then we build the DAG and do topological sort, we build the dependency graph from the registry and compute a safe execution order, skipping cache hits and executing everything in one go.
+
+## Making it Hermetic
+
+Ok so now we know how to graph out orchestration and what tools/data structures we can use to make this all work. But how do we make it hermetic?
+
+Before we go into the gritty details, we should define some context. npm packages run different languages (like our rust compiler) via a binary, which is just a compiled program packaged into a single file like a .exe. The binary is precompiled and runs super fast, you get the idea.
+
+### The Unified Binary Solution
+
+So how do we have a unified binary then? We quickly run into issues for the average user when we consider that they might not have Rust installed, and different OS and CPU combos can create weird execution with binaries, overall does not sound fun to implement and test. This is where **WebAssembly, or WASM** comes in. WASM gives us a portable, precompiled binary that runs on any platform inside a Node process. So we can just compile the Rust generator into a `.wasm` file, then Node will run it via JS wrapper within the npm package, and boom, we have complete coverage over every OS and CPU. Now if a user wants to develop Tinder for Horses on Yocto Linux running on secret technology forbidden Uzbekistani 128 bit CPU architecture, they can do so and we don't have to test and fix random edge cases or juggle messy testing pipelines. Overall the set up seems pretty easy, just need to compile the rust generator on wasm and we're good to go, our JS wrapper will handle the rest. 
+
+And that's orchestration in a nutshell. Obviously up to change as I implement this but in terms of npm orchestration, this is **usually** how orchestration is done and it's what I intend to implement. 
```

Comment:
It seems like the main focus of this initial issue on NPM orchestration is:
1. NPM and Rust to WASM production pipeline (builds publish an NPM package which houses our generator WASM)
2. After including the `cloesce` package to a TS project, running `npm cloesce run` should extract + generate
3. After running `npm cloesce run`, one should either get errors, or be able to just run `wrangler run dev`
4. A super naive setup to just run `npm cloesce run` on any change anywhere in the models dir (wrangler already has this somewhere, I'm sure we can tap into it)

Can you touch on the that first point a bit more? I'm totally unfamiliar with publishing NPM packages, and I haven't ever made a production build pipeline with GitHub actions before.